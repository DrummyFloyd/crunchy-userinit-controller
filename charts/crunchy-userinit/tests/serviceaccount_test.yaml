# yaml-language-server: $schema=https://raw.githubusercontent.com/helm-unittest/helm-unittest/main/schema/helm-testsuite.json
suite: Service Account
templates:
  - serviceaccount.yaml
release:
  name: test
  namespace: test-ns
chart:
  version: 0.0.0
  appVersion: 0.0.0
set:
  image:
    repository: ghcr.io/drummyfloyd/crunchy-userinit-controller
    tag: 0.0.0
tests:
  - it: should match snapshot with defautl
    asserts:
      - matchSnapshot: {}
  - it: should add annotations
    set:
      serviceAccount:
        annotations:
          foo: bar
    asserts:
      - equal:
          path: metadata.annotations
          value:
            foo: bar
  - it: should not create sa
    set:
      serviceAccount:
        create: false
    asserts:
      - hasDocuments:
          count: 0
  - it: should not automount sa
    set:
      serviceAccount:
        automount: false
    asserts:
      - equal:
          path: automountServiceAccountToken
          value: false
  - it: should not set annotations
    asserts:
      - notExists:
          path: metadata.annotations
