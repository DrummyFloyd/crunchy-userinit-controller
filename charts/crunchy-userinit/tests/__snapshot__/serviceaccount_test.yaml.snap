should match snapshot with defautl:
  1: |
    apiVersion: v1
    automountServiceAccountToken: true
    kind: ServiceAccount
    metadata:
      labels:
        app.kubernetes.io/instance: test
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: crunchy-userinit-controller
        app.kubernetes.io/version: 0.0.0
        helm.sh/chart: crunchy-userinit-controller-0.0.0
      name: test-crunchy-userinit-controller
      namespace: test-ns
